rxjs-1/speed.js

```diff
@@ -0,0 +1,31 @@
+const { fromEvent } = require('rxjs');
+const { throttleTime, flatMap } = require('rxjs/operators');
+
+const { getRace } = require('./race');
+const { round } = require('./util');
+
+const race = getRace();
+const carNames = race.getCars();
+
+console.log('Participants:', carNames.join('/'));
+
+const ONE_SEC_IN_MS = 1e3
+
+function getCarSpeed(race, carName) {
+  return fromEvent(race, 'data').pipe(
```

Comment:
This Observable never completes. How could you modify it for it to be completed properly?

Reply:
It is possible to force the completion of a `Subscription` by calling the `.complete()` method, like 🔢  
```javascript
const sub = speed$.subscribe({
  next: (speed) => process.stdout.write(`Speed: ${speed}m/s\r`),
});

// i.e.
setTimeout(() => {
  sub.complete()
}, 2000)
```

Reply:
You need to somehow leverage `race` `'end'` event.